api/server/models/user.js

```diff
@@ -0,0 +1,121 @@
+                            if (action_type == 'facebook' || action_type == 'google') {
+                                verification_status = 1;
+                                verification_code = '';
+                            }
+                            var newUser = new User({
+                                verification_status: verification_status,
+                                verification_code: verification_code,
+                                registration_type: action_type,
+                                social_id: social_id,
+                                platform: platform,
+                                device_id: device_id,
+                                token: token,
+                                email_id: email_id,
+                                name: name,
+                                password: hash_password,
+                                last_seen: '',
+                                registration_time: currentTimestamp,
+                                registration_date: currentDate,
+                                registration_date_time: currentDateTimeDay,
+                                profile_image: ''
+                            });
+                            newUser.save(function (err, result) {
+                                if (err) {
+                                    callback(null, 0, err, {});
+                                } else {
+                                    var user_id = result.id;
+                                    var data = {
+                                        user_id: user_id
+                                    };
+                                    if (action_type != 'facebook' && action_type != 'google') {
+                                        data['show_verification'] = 1;
+                                    }
+                                    User.app.models.email.send_email('new_registration', {email: email_id, name: name, verification_code: verification_code}, function () {
+                                        callback(null, 1, 'Successful Registration', data);
+                                    });
+                                }
+                            });
+                        });
+                    }
+                }
+            }
+        });
+    };
+    User.remoteMethod(
+            'register_login', {
+                accepts: [
+                    {arg: 'action', type: 'string'},
+                    {arg: 'action_type', type: 'string'},
+                    {arg: 'social_id', type: 'string'},
+                    {arg: 'platform', type: 'string'},
+                    {arg: 'device_id', type: 'string'},
+                    {arg: 'token', type: 'string'},
+                    {arg: 'email', type: 'string'},
+                    {arg: 'name', type: 'string'},
+                    {arg: 'password', type: 'string'},
+                    {arg: 'currentTimestamp', type: 'string'},
```

Comment:
you should only recieved 'timestamp' from frontend. not Date and time of day. from time stamp you can get day etc

and the type of this should be numeric